### ISSUE TITLE
Default sharing application

### ISSUE THREAD

**Issue by Sekkei** [MuntashirAkon/AppManager#156]
Good day , I think this these features would fit great with AM 
• Let AM set itself as default sharing application to control/debloat the open-with context menu
• Set default application ( also root/advanced defaults like call-in-screen or default package manager )

( Small fix if possible: 
AM shows only the application name not the action it does in open-with menus )

[ Unlikely feature:
Use the default OS style for pop-ups like the package manager install window )

Thank you very much for your time and your free and open source app !

**Comment by MuntashirAkon** [comment]
> Let AM set itself as default sharing application to control/debloat the open-with context menu

Yes. Will be added after #75.

> Set default application ( also root/advanced defaults like call-in-screen or default package manager )

There's no way to set AM (or any third-party app) as the default package manager without implementing all the features that the package manager has. This is an unnecessary overhead.

> AM shows only the application name not the action it does in open-with menus

It depends on device and application. Buggy implementations display app label instead of activity labels.



> Use the default OS style for pop-ups like the package manager install window

Package manager installer window varies between OS to OS and are closed source.

**Comment by Sekkei** [comment]
> There's no way to set AM (or any third-party app) as the default package manager without implementing all the features that the package manager has. This is an unnecessary overhead.

I meant the general default apps settings , set a music player etc and also allow to set more advanced defaults like package manager etc. Not that AM should be set for all these things.
But out of curiosity which feature are missing ? Probably google play / vendor only features ? Except install of a new app as priv-app ( I have seen the systemless issue 👍🏻 ) I didn't run into limitations until now ( but didn't tried to use adb  )


> It depends on device and application. Buggy implementations display app label instead of activity labels.

Oh okay ! So this could be resolved with my first suggestion you accepted👍🏻


> Package manager installer window varies between OS to OS and are closed source.

I meant more getting the default corner roundness ( or hardcode it for the the 4 vendors ) default boarder size, dark mode and there default colors
This would require ~4 xml's , if you want to provide the backend I would style them ?

**Comment by MuntashirAkon** [comment]
> I meant the general default apps settings , set a music player etc and also allow to set more advanced defaults like package manager etc. Not that AM should be set for all these things.

Yes. This could be done. But as AM has offers many features for APKs, it needs to take care of the defaults inside AM itself instead of relying on Android i.e. if AM will have a single option in the open with section but when you open it, it will display more options. But this needs some major works and is, again, related to #75.

> But out of curiosity which feature are missing ? Probably google play / vendor only features ? Except install of a new app as priv-app ( I have seen the systemless issue 👍🏻 ) I didn't run into limitations until now ( but didn't tried to use adb )

Package Manager (PM) is not just a simple app, it has services, receivers that are used by both user and system software. The broadcast messages sent by PM cannot be sent by any other app.

> I meant more getting the default corner roundness ( or hardcode it for the the ~8 vendors ) default boarder size, dark mode and there default colors

AM is made in a way to ensure that it preserves its own styling (therefore, consistency) everywhere including x86 devices, tablets, TVs and regardless of Android versions. For designing, I used the material design (but not the material colours!) guidelines which almost no vendor/OEM (or even AOSP) follows. Therefore, using OS default design is not a preferred option. Besides, most vendors and APIs do not provide any means to get these values properly. Most of the cases they will only return the default values (as opposed to customised values). For instance, if you have changed the default accent colour, user apps cannot detect them because they are not available via any public API.

> This would require ~8 xml's , if you want to provide the backend I would style them ?

I cannot help you on that matter, I'm afraid. If you have worked with material design library, you would have known how difficult it is to add custom styling. Another problem is that I would need to use themes which means the entire codebase is needed to be altered to support custom theming and will break consistency.

**Comment by MuntashirAkon** [comment]
AM can be set as the default package manager from v2.5.20. However, this will not work for apps that use `PackageInstaller.Session` (there's nothing I can do about it).

> Let AM set itself as default sharing application to control/debloat the open-with context menu

You can choose to open APK files using Interceptor where you can select which apps to open the APK files.

---
_I'm closing this issue. If you have any further requests regarding this topic, either reopen it or open a new issue._
